pkg/api/http_server.go

```diff
@@ -181,6 +187,39 @@ func (hs *HTTPServer) listenAndServeTLS(certfile, keyfile string) error {
 	return hs.httpSrv.ListenAndServeTLS(setting.CertFile, setting.KeyFile)
 }
 
+func (hs *HTTPServer) listenAndServeH2TLS(certfile, keyfile string) error {
+	if certfile == "" {
+		return fmt.Errorf("cert_file cannot be empty when using HTTP2")
+	}
+
+	if keyfile == "" {
+		return fmt.Errorf("cert_key cannot be empty when using HTTP2")
+	}
+
+	if _, err := os.Stat(setting.CertFile); os.IsNotExist(err) {
+		return fmt.Errorf(`Cannot find SSL cert_file at %v`, setting.CertFile)
+	}
+
+	if _, err := os.Stat(setting.KeyFile); os.IsNotExist(err) {
+		return fmt.Errorf(`Cannot find SSL key_file at %v`, setting.KeyFile)
+	}
+
+	tlsCfg := &tls.Config{
+		MinVersion:               tls.VersionTLS12,
+		PreferServerCipherSuites: false,
+		CipherSuites: []uint16{
+			tls.TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,
```

Comment:
I only tagged myself as a reviewer - as I wanted to understand why the cipher-suite is so narrow. I have not been up to speed with the state of HTTP/2 but thinking whenever this would introduce compatibility issues for existing installations.

I understand we don't want it to be too broad due to security risks, but three seems a bit on the shorter end of the stick.

I don't need a direct answer if we have reasons let's just document them as part of the PR description.

Reply:
No reason in particular as to why it's so narrow - purely a consequence of this PR being based on https://github.com/grafana/grafana/pull/13105. I left it unchanged as I wanted to hear from others if they thought, say, Mozilla's recommendations re: preferred ciphers should be followed before committing it to code.

Reply:
For reference I think https://wiki.mozilla.org/Security/Server_Side_TLS is the Mozilla recommendation